dpppt-backend-sdk/dpppt-backend-sdk-data/src/main/java/org/dpppt/backend/sdk/data/gaen/JDBCGAENDataServiceImpl.java

```diff
@@ -63,7 +63,12 @@ public void upsertExposeesDelayed(List<GaenKey> gaenKeys, OffsetDateTime delayed
 		}
 		var parameterList = new ArrayList<MapSqlParameterSource>();
 		var nowMillis = System.currentTimeMillis();
-		//if delayedReceivedAt is supplied use it
+		// if delayedReceivedAt is supplied use it
+		// else we calculate the last moment of the  current  bucket like this:
```

Comment:
Proposition:
```
else calculate what the last millisecond before the next `releaseBucketDuration` will be.
```

Reply:
you mean `the last moment of the current bucket` ;) granted the formulation could be better :D

Reply:
Yes, in fact. Just writing out the formula doesn't tell me _what_ it's supposed to do. And as you see, I got it wrong, even with the formula on multiple lines ;)

In fact I thought about it later and saw my error, but was off for the day...